app/models/shelter.rb

```diff
@@ -31,4 +31,12 @@ def alphabetical_pets
   def shelter_pets_filtered_by_age(age_filter)
     adoptable_pets.where("age >= ?", age_filter)
   end
+
+  def self.reverse_alphabetical  
+    find_by_sql("SELECT * FROM shelters ORDER BY name DESC")
+  end
+
+  def self.pending_apps
+    joins(pets: :applications).where("applications.status = '1'").distinct
```

Comment:
This looks great! Another option here that I'm sharing for fun (and to help with the next project!)
You can set up additional AR associations in order to make this join a bit easier. I call it "mooching off the associations of your children". Pets has associations with applications, and pets belong to shelters. So, shelters can take advantage of pets' associations. If you add the following to this Shelter model:
`has_many :applications, through: :pets`
then you can do this:
`Shelter.joins(:applications)` directly!